doc/source/programmingguide.rst

```diff
@@ -567,6 +567,74 @@ by setting the environement variable ``KOKKOS_TOOLS_LIBS`` to the tool path, for
   export KOKKOS_TOOLS_LIBS=<kokkos-tools-bin>/profiling/space-time-stack/libkp_space_time_stack.so
 
 
+.. _defensiveProgramming::
+
+Defensive Programming
+~~~~~~~~~~~~~~~~~~~~~
+
+``idefix.hpp`` defines useful function-like macros to program defensively and create
+informative error messages and warnings.
+
+First and foremost, ``IDEFIX_ERROR`` and ``IDEFIX_WARNING`` can be used in host space.
+
+.. code-block:: c++
+
+  #include "idefix.hpp"
+
+  #ifdef HAVE_ENERGY
+  IDEFIX_WARNING("This setup is not tested with HAVE_ENERGY defined");
+  #endif
+
+  real boxSize {-1};
+
+  // ... determine boxSize at runtime from parameter file
+
+  if(boxSize<1e-5) {
+    IDEFIX_ERROR("This setup requires a minimal box size of 1e-5");
+  }
+
+
+``idefix.hpp`` also defines the more advanced ``RUNTIME_CHECK_HOST`` and
+``RUNTIME_CHECK_KERNEL`` macros, which are useful to define arbitrary sanity checks at
+runtime in host space and within kernels respectively, together with a nicely formatted
+error message.
+
+Both macros take a condition (a boolean expression that *should* evaluate to ``true`` at
+runtime), and an error message. Additional arguments may be supplemented to the error
+message using string interpolation. As an illustrative example, here's how they can be
```

Comment:
I should also mention here that, because string interpolation isn't possible within GPU kernels, a default (simpler) error message is also required for `RUNTIME_CHECK_KERNEL`